### PR TITLE
Fix parameter encoding and disable autocomplete for password

### PR DESCRIPTION
**Pull request by ChariniNana**

### CODE REVIEW

**Comment by Niranjan-K** [comment]
Please change the commit message and PR subject to imperative-style.

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 1]
```diff
@@ -536,17 +537,17 @@ function displayPasswordField() {
...
+        	<input id="dsName" name=dsName class="longInput" value="<%=Encode.forHtml(dataSourceName) %>">
         <%} %>
         <input type="hidden" id="editMode" name="editMode" value="<%=editMode %>"/>
     </td>
 </tr>
 <tr>
     <td style="width:170px;"><fmt:message key="description"/></td>
     <td align="left">
-        <input id="description" name="description" class="longInput" value="<%=description %>" />
+        <input id="description" name="description" class="longInput" value="<%=Encode.forHtml(description) %>" />
```
Comment (Niranjan-K) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: can the description value be null? if so, wouldn't this throw an exception?
can we check for all the encoding values here?

Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 2]
```diff
@@ -700,26 +701,26 @@ function displayPasswordField() {
 <tr>
     <td><fmt:message key="driver"/><span class='required'>*</span></td>
     <td align="left">
-        <input id="driver" name="driver" class="longInput" value="<%=driverClassName %>"/>
+        <input id="driver" name="driver" class="longInput" value="<%=Encode.forHtml(driverClassName) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Mandatory field. Cannot check for null values

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 3]
```diff
@@ -700,26 +701,26 @@ function displayPasswordField() {
...
     </td>
 </tr>
 <tr>
     <td><fmt:message key="url"/><span class='required'>*</span></td>
     <td align="left">
-        <input id="url" name="url" class="longInput" value="<%=url %>"/>
+        <input id="url" name="url" class="longInput" value="<%=Encode.forHtml(url) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Mandatory field. Cannot check for null values

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 4]
```diff
@@ -700,26 +701,26 @@ function displayPasswordField() {
...
     </td>
 </tr>
 <tr>
     <td><fmt:message key="user.name"/></td>
     <td align="left">
-        <input id="username" name="username" class="longInput" value="<%=username %>"/>
+        <input id="username" name="username" class="longInput" value="<%=Encode.forHtml(username) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 5]
```diff
@@ -700,26 +701,26 @@ function displayPasswordField() {
...
     </td>
 </tr>
 <% if (!editMode) { %>
 	<tr>
     <td><fmt:message key="password"/></td>
     <td align="left">
-        <input id="password" name="password" type="password" class="longInput"/>
+        <input id="password" name="password" type="password" class="longInput" autocomplete="off"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 6]
```diff
@@ -732,15 +733,15 @@ function displayPasswordField() {
 	
 <tr id="newPasswordRow" style="display:none">
 	<td><label for="changePassword"><fmt:message  key="password"/></label></td>
-	<td><input type="password" id="newPassword" name="newPassword" class="longInput"/></td>
+	<td><input type="password" id="newPassword" name="newPassword" class="longInput" autocomplete="off"/></td>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 7]
```diff
@@ -732,15 +733,15 @@ function displayPasswordField() {
...
 </tr>
 	
 
 <% } else if (NDataSourceClientConstants.RDBMS_EXTERNAL_DATASOURCE_PROVIDER.equals(dsProvider)){ %>
 <tr>
     <td><fmt:message key="datasource.className"/><span class='required'>*</span></td>
     <td align="left">
-        <input id="dsclassname" name="dsclassname" class="longInput" value="<%=dataSourceclassName %>"/>
+        <input id="dsclassname" name="dsclassname" class="longInput" value="<%=Encode.forHtml(dataSourceclassName) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Mandatory field. Cannot check for null values

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 8]
```diff
@@ -790,7 +791,7 @@ function displayPasswordField() {
         	<tr>
     <td style="width:170px;"><fmt:message key="jndi.name"/></td>
     <td align="left">
-    	<input id="jndiname" name="jndiname" class="longInput" value="<%=jndiConfigName %>" />
+    	<input id="jndiname" name="jndiname" class="longInput" value="<%=Encode.forHtml(jndiConfigName) %>" />
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 9]
```diff
@@ -915,7 +916,7 @@ function displayPasswordField() {
 			<tr>
 			    <td><fmt:message key="default.catalogn"/></td>
 			    <td align="left">
-			        <input id="defaultCatalog" name="defaultCatalog" type="text" value="<%=defaultCatalog %>"/>
+			        <input id="defaultCatalog" name="defaultCatalog" type="text" value="<%=Encode.forHtml(defaultCatalog) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 10]
```diff
@@ -1024,12 +1025,12 @@ function displayPasswordField() {
 			<tr>
 			    <td><fmt:message key="validation.query"/></td>
 			    <td align="left">
-			    <input id="validationquery" name="validationquery" type="text" value="<%=validationquery %>"/>
+			    <input id="validationquery" name="validationquery" type="text" value="<%=Encode.forHtml(validationquery) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 11]
```diff
@@ -1024,12 +1025,12 @@ function displayPasswordField() {
...
 			</tr>
 			<tr>
 			    <td><fmt:message key="validation.class.name"/></td>
 			    <td align="left">
-			    <input id="validatorClassName" name="validatorClassName" type="text" value="<%=validatorClassName %>"/>
+			    <input id="validatorClassName" name="validatorClassName" type="text" value="<%=Encode.forHtml(validatorClassName) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 12]
```diff
@@ -1124,17 +1125,17 @@ function displayPasswordField() {
 			<tr> 
 			    <td><fmt:message key="connection.properties"/></td>
 			    <td align="left">
-			    <input id="connectionProperties" name="connectionProperties" type="text" value="<%=connectionProperties %>"/>
+			    <input id="connectionProperties" name="connectionProperties" type="text" value="<%=Encode.forHtml(connectionProperties) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 13]
```diff
@@ -1124,17 +1125,17 @@ function displayPasswordField() {
...
 			</tr>
 			<tr>
     			<td style="width:230px;"><fmt:message key="init.sql"/></td>
     			<td align="left">
-    				<input id="initSQL" name="initSQL" type="text" value="<%=initSQL %>"/>
+    				<input id="initSQL" name="initSQL" type="text" value="<%=Encode.forHtml(initSQL) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

File: components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp [hunk 14]
```diff
@@ -1124,17 +1125,17 @@ function displayPasswordField() {
...
 			</tr>
 			<tr>
     			<td><fmt:message key="jdbc.interceptors"/></td>
     			<td align="left">
-    				<input id="jdbcInterceptors" name="jdbcInterceptors" type="text" value="<%=jdbcInterceptors %>"/>
+    				<input id="jdbcInterceptors" name="jdbcInterceptors" type="text" value="<%=Encode.forHtml(jdbcInterceptors) %>"/>
```
Comment (ChariniNana) [inline components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/resources/web/ndatasource/newdatasource.jsp]: Checked for null value. Doesn't throw an error

**Pull request merged**